staging/src/k8s.io/api/resource/v1alpha3/types.go

```diff
@@ -0,0 +1,998 @@
+const AllocationResultsMaxSize = 32
+
+// RequestAllocationResult contains the allocation result for one request.
+type RequestAllocationResult struct {
+	// RequestName identifies the request in the claim which caused this
+	// device to be allocated. Multiple devices may have been allocated
+	// per request.
+	//
+	// +required
+	RequestName string `json:"requestName" protobuf:"bytes,1,name=requestName"`
+
+	// DriverName specifies the name of the DRA driver whose kubelet
+	// plugin should be invoked to process the allocation once the claim is
+	// needed on a node.
+	//
+	// Must be a DNS subdomain and should end with a DNS domain owned by the
+	// vendor of the driver.
+	//
+	// +required
+	DriverName string `json:"driverName" protobuf:"bytes,2,name=driverName"`
+
+	// This name together with the driver name and the device name field
+	// identify which device was allocated (`<driver name>/<pool name>/<device name>`).
+	//
+	// Must not be longer than 253 characters and may contain one or more
+	// DNS sub-domains separated by slashes.
+	//
+	// +required
+	PoolName string `json:"poolName" protobuf:"bytes,3,name=poolName"`
+
+	// DeviceName references one device instance via its name in the driver's
+	// resource pool. It must be a DNS label.
+	//
+	// +required
+	DeviceName string `json:"deviceName" protobuf:"bytes,4,name=deviceName"`
+}
+
+// AllocationConfiguration gets embedded in an AllocationResult.
+type AllocationConfiguration struct {
+	// Admins is true if the source of the configuration was a class and thus
+	// not something that a normal user would have been able to set.
+	Admin bool `json:"admin,omitempty" protobuf:"bytes,1,opt,name=admin"`
+
+	// The configuration applies to devices in these requests.
+	//
+	// If empty, the configuration applies to all devices in the claim.
+	//
+	// +optional
+	// +listType=atomic
+	RequestNames []string `json:"requestNames,omitempty" protobuf:"bytes,2,opt,name=requestNames"`
+
+	Configuration `json:",inline" protobuf:"bytes,3,name=configuration"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:prerelease-lifecycle-gen:introduced=1.26
+
+// ResourceClaimList is a collection of claims.
+type ResourceClaimList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Items is the list of resource claims.
+	Items []ResourceClaim `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:prerelease-lifecycle-gen:introduced=1.26
+
+// PodSchedulingContext objects hold information that is needed to schedule
+// a Pod with ResourceClaims that use "WaitForFirstConsumer" allocation
+// mode.
+//
+// This is an alpha type and requires enabling the DynamicResourceAllocation
+// feature gate.
+type PodSchedulingContext struct {
```

Comment:
Is this all governed by gate "DRAControlPlaneController" now?

Reply:
Yes. Fixed.